src/main/java/edu/kit/provideq/toolbox/api/SolveRouter.java

```diff
@@ -142,6 +135,146 @@ private Mono<ServerResponse> handleSolutionRouteForMetaSolver(MetaSolver<?, ?, ?
+                    .example(getExampleInvalid(metaSolver))
+                    .array(arraySchemaBuilder().schema(
+                            schemaBuilder().implementation(Solution.class))));
+  }
+
+  private static org.springdoc.core.fn.builders.exampleobject.Builder getExampleSolved(
+          MetaSolver<?, ?, ?> metaSolver) {
+    return getExampleOk(
+            "Solved",
+            metaSolver,
+            solution -> {
+              solution.setSolutionData("Solution data to solve the problem");
+              solution.complete();
+            });
+  }
+
+  private static org.springdoc.core.fn.builders.exampleobject.Builder getExampleInvalid(
+          MetaSolver<?, ?, ?> metaSolver) {
+    return getExampleOk(
+            "Error",
+            metaSolver,
+            solution -> {
+              solution.setDebugData("Some error occurred");
+              solution.abort();
+            });
+  }
+
+  private static org.springdoc.core.fn.builders.exampleobject.Builder getExampleOk(
+          String exampleName,
+          MetaSolver<?, ?, ?> metaSolver,
+          Consumer<Solution<String>> solutionModifier) {
+    // Prepare a solved solution with some example data
+    var solvedSolution = new Solution<String>(42);
+    solvedSolution.setExecutionMilliseconds(42);
+    metaSolver.getAllSolvers().stream()
+            .findFirst()
+            .ifPresent(solver -> solvedSolution.setSolverName(solver.getName()));
+    solutionModifier.accept(solvedSolution);
+
+    // Convert the solution to a string
+    String solvedSolutionString;
+    try {
+      solvedSolutionString = new ObjectMapper().writeValueAsString(solvedSolution);
+    } catch (JsonProcessingException e) {
+      solvedSolutionString = "Error: example could not be parsed";
+    }
+
+    // Build the example
+    return org.springdoc.core.fn.builders.exampleobject.Builder
+            .exampleOjectBuilder()
+            .name(exampleName)
+            .description("The problem was solved successfully.")
+            .value(solvedSolutionString);
+  }
+
+  private static org.springdoc.core.fn.builders.apiresponse.Builder getResponseNotFound() {
+    return responseBuilder()
+            .responseCode(String.valueOf(HttpStatus.NOT_FOUND.value()));
+  }
+
+  private org.springdoc.core.fn.builders.content.Builder getRequestContent(
+          MetaSolver<?, ?, ?> metaSolver) {
+    String content = metaSolver.getExampleProblems()
+            .stream().findFirst()
+            .map(e -> {
+              if (e instanceof String) {
+                return (String) e;
+              }
+
+              try {
+                return new ObjectMapper().writeValueAsString(e);
+              } catch (JsonProcessingException exception) {
+                return "Error: example could not be parsed";
+              }
+            })
+            .orElse("Error: no example available");
+
+    var request = new SolveRequest<String>();
```

Comment:
I think, `SolveRequest`s can have different content types than `String`

Reply:
Sure, but for the purpose of this example, I convert all examples to their string representation, see the initialization of `String content` in line 216.

Reply:
I don't think converting to String keeps the structure of the request. If we add requests with a (non-string) object content type, converting those to json, the example request will be something like this:

```json
{
  "requestContent": "{\"x\": 42, \"y\": \"[kœri]werk®\"}"
}
```

while the actual request is expected to look like this:

```json
{
  "requestContent": {
    "x": 42,
    "y": "[kœri]werk®"
  }
}
```

Reply:
Ah yes, this definitely makes sense. Sadly though, using a SolveRequest of type `Object` doesn't help with the format that we get in the final JSON representation in swagger.

Reply:
I think the way it's done now is correct. If we only specify the content type as object in this method, the JSON serializer will still serialize the specific type of the object instance